contents/data_vis.md

```diff
@@ -523,12 +541,193 @@ Next, we will dive into how to manage and control **layouts**.
+
+> - `colgap!(fig.layout, col, separation)`
+> - `rowgap!(fig.layout, row, separation)`
+>
+> _Column gap_, if `col` is given then the gap will be applied to that column. 
+>_Row gap_ ,if `row` is given then the gap will be applied to that row. 
+
+Also, we will see how to put content into the **protrusions**, _i.e._ the space reserved for _title, x - (ticks,label), y - (ticks,label)_. 
+We do this by plotting into `fig[i,j, protrusion]` where _`protrusion`_ can be `Left()`, `Right()`, `Bottom()` and `Top()`, 
+or for each corner `TopLeft()`, `TopRight()`, `BottomRight()`, `BottomLeft()`. 
+See below how these options are being used. 
+
+```jl
+@sco JDS.first_layout_fixed()
+```
+
+Here, probably having the label `(a)` in the `TopLeft()` is not necessary, this will make sense for more than two plots. 
+For our next example let's use the previous tools and some more to create a more complex figure. 
+
+You can hide decorations and axis's spines with
+
+> - `hidedecorations!(ax; kwargs...)`
+> - `hidexdecorations!(ax; kwargs...)`
+> - `hideydecorations!(ax; kwargs...)`
+> - `hidespines!(ax; kwargs...)`
+
+Remember, we can always ask for help to see what kind of arguments we can use, e.g.,
+
+```jl
+s = """
+    help(hidespines!)
+    """
+sco(s)
+```
+
+Or, for decorations
+
+```jl
+s = """
+    help(hidedecorations!)
+    """
+sco(s)
+```
+
+For elements that **you don't want to hide**, just pass them with `false`, i.e. `hideydecorations!(ax, ticks = false, grid = false)`.
+
+
+Synchronizing your `Axis` is done via
+
+> - `linkaxes! linkyaxes! linkxaxes!`
+>
+> This could be useful when shared axis are desired. Another way of getting shared axis will be by setting `limits!`.
+
+Setting `limits` at once or independenly for each axis is done by calling 
+
+> - `limits!(ax; l,r,b,t)`
+> You can also do `ylims!(low, high)` or `xlims!(low, high)`, and even open ones by doing `ylims!(low=0)` or `xlims!(high=1)`.
+
+Now, the example:
+
+```jl
+@sco JDS.complex_layout_double_axis()
+```
+
+So, now our `Colorbar` needs to be horizontal and the bar ticks be in the lower part. 
+This is done by setting `vertical = false` and `flipaxis = false`. 
+Also, note that we can call many `Axis` into `fig`, or even `Colorbar`'s and `Legend`'s, and then afterwards build the layout. 
+
+Another common layout is a grid of squares for heatmaps. See next example. 
+
+```jl
+@sco JDS.squares_layout()
+```
+
+Where, all labels are in the **protrusions** and each `Axis` has an `AspectData()` ratio. 
+The `Colorbar` is located in the third column and extpands from row 1 up to row 2. 
+
+The next case uses the so called `Mixed()` **alignmode**, which is specially useful when dealing with large empty spaces between `Axis` due to long ticks. 
+
+```jl
+@sco JDS.mixed_mode_layout()
+```
+
+Here, the argument `alignmode =Mixed(bottom=0)` it's shifting the bounding box to the bottom, so that this will align with the panel on the left filling the space. 
```

Comment:
```suggestion
Here, the argument `alignmode = Mixed(bottom=0)` is shifting the bounding box to the bottom, so that this will align with the panel on the left filling the space. 
```